flink-ml-lib/src/main/java/org/apache/flink/ml/classification/linear/LogisticRegressionModel.java

```diff
@@ -0,0 +1,213 @@
+import java.util.HashMap;
+import java.util.Map;
+
+/** This class implements {@link Model} for {@link LogisticRegression}. */
+public class LogisticRegressionModel
+        implements Model<LogisticRegressionModel>,
+                LogisticRegressionModelParams<LogisticRegressionModel> {
+
+    private Map<Param<?>, Object> paramMap = new HashMap<>();
+
+    private Table model;
+
+    public LogisticRegressionModel() {
+        ParamUtils.initializeMapWithDefaultValues(this.paramMap, this);
+    }
+
+    @Override
+    public Map<Param<?>, Object> getParamMap() {
+        return paramMap;
+    }
+
+    @Override
+    public void save(String path) throws IOException {
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) model).getTableEnvironment();
+        String dataPath = ReadWriteUtils.getDataPath(path);
+        FileSink<LogisticRegressionModelData> sink =
+                FileSink.forRowFormat(new Path(dataPath), new LogisticRegressionModelDataEncoder())
+                        .withRollingPolicy(OnCheckpointRollingPolicy.build())
+                        .withBucketAssigner(new BasePathBucketAssigner<>())
+                        .build();
+        ReadWriteUtils.saveMetadata(this, path);
+        tEnv.toDataStream(model)
+                .map(x -> (LogisticRegressionModelData) x.getField(0))
+                .sinkTo(sink)
+                .setParallelism(1);
+    }
+
+    public static LogisticRegressionModel load(StreamExecutionEnvironment env, String path)
+            throws IOException {
+        StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
+        Source<LogisticRegressionModelData, ?, ?> source =
+                FileSource.forRecordStreamFormat(
+                                new LogisticRegressionModelDataStreamFormat(),
+                                ReadWriteUtils.getDataPaths(path))
+                        .build();
+        LogisticRegressionModel model = ReadWriteUtils.loadStageParam(path);
+        DataStream<LogisticRegressionModelData> modelData =
+                env.fromSource(source, WatermarkStrategy.noWatermarks(), "modelData");
+        model.setModelData(tEnv.fromDataStream(modelData));
+        return model;
+    }
+
+    @Override
+    public LogisticRegressionModel setModelData(Table... inputs) {
+        model = inputs[0];
+        return this;
+    }
+
+    @Override
+    public Table[] getModelData() {
+        return new Table[] {model};
+    }
+
+    @Override
+    public Table[] transform(Table... inputs) {
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) inputs[0]).getTableEnvironment();
+
+        DataStream<Row> data = tEnv.toDataStream(inputs[0]);
+        final String broadcastModelName = "broadcastModel";
+        DataStream<LogisticRegressionModelData> modelData =
+                tEnv.toDataStream(model).map(x -> (LogisticRegressionModelData) x.getField(0));
```

Comment:
Could you add the static `LogisticRegressionModelData::getModelDataStream` to do this conversion? The NaiveBayes PR  does this and I find this to be useful. The reason is that this approach centralizes the knowledge the the model data schema in one class. And in the future when we provide util methods for users to manually construct and set model data,  those methods can be easily added in e.g. `LogisticRegressionModelData`.

Reply:
Nice improvement!

We can actually go one more step --- 

- add two more static method in `ReadWriteUtils`, i.e., 
`static void saveModelData(Table model, String path, ModelEncoder encoder)` and `static Table loadModelData(StreamExecutionEnvironment env, String path, ModelDecoder decoder)`.
- add two more static method in `LogisticRegressionModelData`, i.e., `getModelDataEncoder` and `getModelDataDecoder`.

In this case, we can cover everything about model data in one class. We may probably come up with a new interface here.

Reply:
In general we only use interface to programmatically force a rule that is important to users. The APIs we are discussing here is for algorithm developers. It is probably simpler not to use interface for now.

I am not sure if the proposed "further step" is the way to go before seeing the code. My intuition is that going one step further may be unnecessary. Feel free to just follow Naive Bayes approach, or go one step further if you are convinced it will be better.